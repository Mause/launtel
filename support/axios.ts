import axios from "axios";

const create = axios.create;

export default { create };
