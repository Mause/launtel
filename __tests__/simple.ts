import { createServer } from "vercel-node-server";
import listen from "test-listen";
import axios from "axios";
import routeUnderTest, { axios as axiosUp } from "../api/up";
import { Server } from "net";
import moxios from "moxios";
import { sign } from "jsonwebtoken";

let server: Server;
let url: string;

const SECRET = "SECRET";
process.env.JWT_SECRET = SECRET;

beforeAll(async () => {
  server = createServer(routeUnderTest);
  url = await listen(server);
  moxios.install(axiosUp);
});

afterAll(() => {
  server.close();
  moxios.uninstall(axiosUp);
});

it("should return the expected response", async () => {
  moxios.stubOnce("GET", /.*/, {
    status: 200,
    response: {
      data: [
        {
          attributes: {
            amount: { valueInBaseUnits: 1500 },
            rawText: "Internet bill",
          },
        },
      ],
      links: {
        next: "https://up/next",
      },
    },
  });
  const response = await axios.get(url, {
    headers: {
      Authorization: "Bearer " + sign({ aud: "authenticated" }, SECRET),
    },
  });
  expect(response.status).toBe(200);
  expect(response.data).toMatchSnapshot();
});
