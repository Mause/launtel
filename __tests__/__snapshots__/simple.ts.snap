// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return the expected response 1`] = `
Object {
  "items": Array [
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
    Object {
      "attributes": Object {
        "amount": Object {
          "valueInBaseUnits": 1500,
        },
        "rawText": "Internet bill",
      },
    },
  ],
}
`;
